### PR TITLE
Adding upgrading doc for plugins to move from Elasticsearch to OpenSe…

### LINKED ISSUE #2: Document an upgrade path for plugins
HowTo make existing plugins compatible with OpenSearch, whether migrating or upgrading.

### PR DESCRIPTION
**Pull request by saratvemulapalli**
Signed-off-by: Sarat Vemulapalli <vemulapallisarat@gmail.com>

### Description
Adding upgrading doc for plugins to move from Elasticsearch to OpenSearch

### Issues Resolved
#2 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+## Upgrading Plugins to OpenSearch/OpenSearch Dashboards
+
+For the plugins to work with OpenSearch here are all the steps you can take and get them building, passing all the tests.
```
Comment (dblock) [inline UPGRADING.md:3]: Nit on the English, switch 

"These are all the steps to upgrade plugins to work with OpenSearch and OpenSearch Dashboards, including building, and passing all tests."

File: UPGRADING.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+### Upgrading
+We will upgrade the plugins to use OpenSearch/Dashboards 1.0.0.
+
+#### OpenSearch Plugins
+
+1. Consume artifacts from OpenSearch by publishing them locally. Please see [Readme](./README.md#building-with-opensearch)
```
Comment (dblock) [inline UPGRADING.md:10]: Missing period.

File: UPGRADING.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+2. Change the namespaces from `org.elasticsearch` to `org.opensearch`. Use the naming convention below.
+3. Consuming dependencies with OpenSearch/Dashboards 1.0.0. 
+   * For dependencies `securemock`, `mocksocket` and `jna-build` please continue to use `org.elasticsearch` namespace.
+4. Build and test your plugin.
+5. Please report all run time failures of OpenSearch to [OpenSearch Issues](http://github.com/opensearch-project/opensearch/issues) and
```
Comment (dblock) [inline UPGRADING.md:15]: remove "Please"

Comment (dblock) [reply UPGRADING.md:15]: runtime is one word

File: UPGRADING.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+1. Consume artifacts from OpenSearch by publishing them locally. Please see [Readme](./README.md#building-with-opensearch)
+2. Change the namespaces from `org.elasticsearch` to `org.opensearch`. Use the naming convention below.
+3. Consuming dependencies with OpenSearch/Dashboards 1.0.0. 
+   * For dependencies `securemock`, `mocksocket` and `jna-build` please continue to use `org.elasticsearch` namespace.
+4. Build and test your plugin.
+5. Please report all run time failures of OpenSearch to [OpenSearch Issues](http://github.com/opensearch-project/opensearch/issues) and 
+runtime failures of plugins in [OpenSearch plugin Issues](https://github.com/opensearch-project/opensearch-plugins/issues).
+
+Here is an example for the changes which were done for OpenSearch plugin:
```
Comment (dblock) [inline UPGRADING.md:18]: Period :)

**Review by dblock — APPROVED** [review]
Only nits. 

Also link this from README.

File: UPGRADING.md (hunk 1 above)
Comment (saratvemulapalli) [reply UPGRADING.md:3]: Got it, makes sense.

**Pull request merged**